src/stackwalk.c

```diff
@@ -0,0 +1,411 @@
+    }
+    bt = jl_alloc_array_1d(array_ptr_void_type, jl_bt_size);
+    memcpy(bt->data, jl_bt_data, jl_bt_size * sizeof(void*));
+    JL_GC_POP();
+    return (jl_value_t*)bt;
+}
+
+
+
+#if defined(_OS_WINDOWS_)
+#ifdef _CPU_X86_64_
+static UNWIND_HISTORY_TABLE HistoryTable;
+#else
+static struct {
+    DWORD64 dwAddr;
+    DWORD64 ImageBase;
+} HistoryTable;
+#endif
+static PVOID CALLBACK JuliaFunctionTableAccess64(
+        _In_  HANDLE hProcess,
+        _In_  DWORD64 AddrBase)
+{
+    //jl_printf(JL_STDOUT, "lookup %d\n", AddrBase);
+#ifdef _CPU_X86_64_
+    DWORD64 ImageBase;
+    PRUNTIME_FUNCTION fn = RtlLookupFunctionEntry(AddrBase, &ImageBase, &HistoryTable);
+    if (fn) return fn;
+    if (jl_in_stackwalk) {
+        return 0;
+    }
+    jl_in_stackwalk = 1;
+    PVOID ftable = SymFunctionTableAccess64(hProcess, AddrBase);
+    jl_in_stackwalk = 0;
+    return ftable;
+#else
+    return SymFunctionTableAccess64(hProcess, AddrBase);
+#endif
+}
+static DWORD64 WINAPI JuliaGetModuleBase64(
+        _In_  HANDLE hProcess,
+        _In_  DWORD64 dwAddr)
+{
+    //jl_printf(JL_STDOUT, "lookup base %d\n", dwAddr);
+#ifdef _CPU_X86_64_
+    DWORD64 ImageBase;
+    PRUNTIME_FUNCTION fn = RtlLookupFunctionEntry(dwAddr, &ImageBase, &HistoryTable);
+    if (fn) return ImageBase;
+    if (jl_in_stackwalk) {
+        return 0;
+    }
+    jl_in_stackwalk = 1;
+    DWORD64 fbase = SymGetModuleBase64(hProcess, dwAddr);
+    jl_in_stackwalk = 0;
+    return fbase;
+#else
+    if (dwAddr == HistoryTable.dwAddr) return HistoryTable.ImageBase;
+    DWORD64 ImageBase = jl_getUnwindInfo(dwAddr);
+    if (ImageBase) {
+        HistoryTable.dwAddr = dwAddr;
+        HistoryTable.ImageBase = ImageBase;
+        return ImageBase;
+    }
+    return SymGetModuleBase64(hProcess, dwAddr);
+#endif
+}
+
+static void jl_unw_get(bt_context_t *context)
+{
+    memset(context, 0, sizeof(*context));
+    RtlCaptureContext(&context);
+}
+
+int needsSymRefreshModuleList;
+BOOL (WINAPI *hSymRefreshModuleList)(HANDLE);
+static int jl_unw_init(bt_cursor_t *cursor, bt_context_t *Context)
+{
+    if (needsSymRefreshModuleList && hSymRefreshModuleList != 0 && !jl_in_stackwalk) {
+        jl_in_stackwalk = 1;
+        hSymRefreshModuleList(GetCurrentProcess());
+        jl_in_stackwalk = 0;
+        needsSymRefreshModuleList = 0;
+    }
+#if !defined(_CPU_X86_64_)
+    if (jl_in_stackwalk) {
+        return 0;
+    }
+    DWORD MachineType = IMAGE_FILE_MACHINE_I386;
+    memset(&cursor->stackframe, 0, sizeof(cursor->stackframe));
+    cursor->stackframe.AddrPC.Offset = Context->Eip;
+    cursor->stackframe.AddrStack.Offset = Context->Esp;
+    cursor->stackframe.AddrFrame.Offset = Context->Ebp;
+    cursor->stackframe.AddrPC.Mode = AddrModeFlat;
+    cursor->stackframe.AddrStack.Mode = AddrModeFlat;
+    cursor->stackframe.AddrFrame.Mode = AddrModeFlat;
+    jl_in_stackwalk = 1;
+#endif
+    cursor->context = Context;
```

Comment:
how is this supposed to work in the 64 bit case where you've done `typedef CONTEXT bt_cursor_t` rather than a struct?